osquery/tables/system/darwin/ssdeep.cpp

```diff
@@ -0,0 +1,100 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include "osquery/tables/system/darwin/ssdeep.h"
+
+#include <boost/filesystem.hpp>
+#include <fuzzy.h>
+#include <osquery/filesystem.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+namespace osquery {
+namespace tables {
+
+void expandFSPathConstraints(QueryContext& context,
+                             std::string path_column_name,
+                             std::set<std::string>& paths) {
+  context.expandConstraints(
+    "path",
+    LIKE,
+    paths,
+    ([&](const std::string& pattern, std::set<std::string>& out) {
+      std::vector<std::string> patterns;
+      auto status =
+        resolveFilePattern(pattern, patterns, GLOB_ALL | GLOB_NO_CANON);
+      if (status.ok()) {
+        for (const auto& resolved : patterns) {
+          out.insert(resolved);
+        }
+      }
+      return status;
+    }));
+}
+
+void genSsdeepForFile(const std::string& path,
+                      const std::string& dir,
+                      QueryData& results) {
+  auto fd = fopen(path.c_str(), "r");
+  char *file_ssdeep_c_str = (char *)malloc(FUZZY_MAX_RESULT);
+  fuzzy_hash_file(fd, file_ssdeep_c_str);
```

Comment:
`fuzzy_*` functions could return error (non-zero return value). So, to avoid any kind of incorrect data in the table it is better to handle `fuzzy_*` errors here. What do you think?